### PR TITLE
Internationalisation

### PR DESCRIPTION
**Pull request by davidwessman**
Started looking at this a bit, smooth switching and the linking works properly, adding `/en` before any path to get english locale.

 - [ ] `Globalize` (translsation for object fields)
 - [ ] `I18n-tasks` implemented properly (perhaps Google API for translation)
 - [ ] Fix comments from @jforberg

### CODE REVIEW

File: app/controllers/application_controller.rb [hunk 1]
```diff
@@ -120,7 +120,8 @@ def set_locale
       end
     end
     I18n.locale = locale
```
Comment (jforberg) [inline app/controllers/application_controller.rb]: This function is quite messy. Is there a clearer way to write it?

If it has to be that messy, please add a comment explaining what it does as that is currently not obvious.

File: app/views/layouts/_topbar.html.erb [hunk 2]
```diff
@@ -1,6 +1,14 @@
 <div class="top">
   <div class="container">
     <ul class="loginbar pull-right">
+      <li>
+        <% if I18n.locale == :sv %>
+        <%= link_to('English', locale: :en) %>
```
Comment (jforberg) [inline app/views/layouts/_topbar.html.erb]: Indent

File: app/views/layouts/_topbar.html.erb [hunk 3]
```diff
@@ -1,6 +1,14 @@
...
+        <% else %>
+        <%= link_to('Svenska', locale: :sv) %>
+      <% end %>
+      </li>
+      <li class="devider"> </li>
```
Comment (jforberg) [inline app/views/layouts/_topbar.html.erb]: divider, not devider

File: config/environments/development.rb [hunk 4]
```diff
@@ -44,7 +44,7 @@
   config.serve_static_files = true
   PUBLIC_URL = 'localhost:3000'
 
-  config.action_view.raise_on_missing_translations = true
+  config.action_view.raise_on_missing_translations = false
```
Comment (jforberg) [inline config/environments/development.rb]: Good. These exceptions are annoying.

**Comment by jforberg** [comment]
Looks good but you are touching some old code which needs a bit of work. This is a good time to do it.

-1

File: app/views/layouts/_topbar.html.erb (hunk 3 above)
Comment (davidwessman) [inline app/views/layouts/_topbar.html.erb]: The fantastic spelling of our bough theme - can change it :)

File: app/controllers/application_controller.rb (hunk 1 above)
Comment (davidwessman) [inline app/controllers/application_controller.rb]: Yep. I just edited this to follow the param-style.
I do not think that we should use cookies for this either.
Will fix.

**Comment by davidwessman** [comment]
@jforberg 
I just started looking at this since we were speaking of it in #248. 
And we were speaking about the google translate fill in and stuff. 
I looked at it a bit more yesterday, it needs an google API costing 20 $ per 2M characters (which seems pretty cheap, could definitely be worth it).

tl:dr;
Not ready for merging - just wanted to show, perhaps shouldn't have made a PR :innocent:

**Comment by jforberg** [comment]
That's not expensive at all. We just have to cache everything to avoid paying twice. That's an interesting problem actually, I'd be interested in looking at it.

It's totally ok to post pull requests just for comments.

**Comment by davidwessman** [comment]
@jforberg And I was looking at using google translate with the https://github.com/glebm/i18n-tasks feature to auto translate the .yml files, meaning that you translate missing keys and automatically save them as static locale files :)

**Comment by jforberg** [comment]
Hmm... You could write a grand caching system which caches the whole page. Then you wouldn't have to bother about I18n at all :) it would be a lot of work though.

**Pull request closed without merging**